src/main/java/org/apache/commons/lang3/text/translate/CharSequenceTranslator.java

```diff
@@ -121,7 +127,8 @@ public final void translate(final CharSequence input, final Writer out) throws I
      * @return CharSequenceTranslator merging this translator with the others
      */
     public final CharSequenceTranslator with(final CharSequenceTranslator... translators) {
-        final CharSequenceTranslator[] newArray = new CharSequenceTranslator[translators.length + 1];
+        @SuppressWarnings("assignment.type.incompatible") // translators.length + 1 has minimum value 1
```

Comment:
Why is this necessary? If this annotation isn't inferred, this is a bug in the Index Checker.

Reply:
It isn't inferred and I think it is due to the imprecision in the Index Checker related to multi-dimensional arrays